cpp/src/fil/infer.cu

```diff
@@ -260,21 +405,35 @@ struct tree_aggregator_t<NITEMS, GROVE_PER_CLASS_FEW_CLASSES> : finalize_block {
     }
     __syncthreads();  // free up per_thread[] margin
 
-    write_best_class_in_block(to_vec(threadIdx.x, acc), num_classes, out,
-                              num_rows);
+    if (num_outputs == 1) {  // will output class
+      auto candidate = to_vec(threadIdx.x, acc);
+      write_best_class_in_block<NITEMS>(
```

Comment:
It's better to do this in a conditional:
```
if (threadIdx.x < num_classes) {
  write_best_class_in_block<NITEMS>(...);
}
```

Reply:
I cannot use a conditional because the function uses __syncthreads() when reducing the candidate classes with an argmax. I have to provide some moot value, but I can keep the extra warps busy reducing {-1, -INFINITY} instead of signalling that this thread does not have a valid candidate.
Another option is to move the tree reduction into write_best_class_in_block.